### DISCUSSION TITLE
How to get client.meshtastic.org to work with the nRF52 over ethernet

### DISCUSSION THREAD

**Discussion by ckambiselis** [meshtastic/Discussions#68] [nRF52]
So, since the flash on the nRF52 is so small that the web client can't be installed on it, how can one get the certificate installed, since TLS can't be turned off on client.meshtastic.org? And not really interested in using the Android client, since this is going to be a base station node. Or is the only way to make it work, to self host the client?

**Comment by ianmcorvidae** [comment]
Hi; you won't be able to get the web client working with an nrf52 device over HTTP regardless of TLS certificates. It's more than just that it can't hold the web client in its flash, it doesn't deliver a web server at all. Android and CLI are able to connect to it by way of the TCP API (which is separate from the HTTP one). I made a proof-of-concept project that serves up an HTTP API and proxies it to an underlying TCP/Serial/BLE connection, which is at https://github.com/ianmcorvidae/meshtastic-http-proxy (but no promises it'll work right, no support guaranteed, etc.). If you were to use that, you'd run the proxy somewhere it can connect to your node and where you can serve it up with a TLS certificate and use it with client.meshtastic.org, or you would serve it up without one and self-host the web client to connect to it. Another option would be to connect with Web Bluetooth (or serial) from a chromium-based browser that supports it, but obviously that wouldn't use the Ethernet connection and might not be an option for you.

**Reply by ckambiselis** [reply]
Ok, maybe I have misunderstood how the Android app connects to it and I'm using the wrong words, but the app has an option to connect using an IP address, which is how I have been able to connect to it and change settings. How can I get this type of connection through Chrome on a PC, if it is some kind of "serial over TCP", going to the serial tab doesn't give an option to enter an IP.

**Reply by ianmcorvidae** [reply]
You can't, without running additional stuff or using a different connection method. The Android app and the web client are using two different methods: the TCP API and the HTTP API, respectively.

Here's the table of the different apps and the ways they're able to connect:

|              | BLE | Serial | TCP | HTTP
|------------|-------|----------|-------|---------|
| Android | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :x: |
| iOS | :heavy_check_mark: | :x: | :x: | :x: |
| CLI | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :x: |
| Web | :heavy_check_mark: * | :heavy_check_mark: * | :x: | :heavy_check_mark: |

(* web can only do BLE/Serial on browsers supporting the Web Bluetooth and Web Serial APIs)

[edited to add the following for extra info for folks]

Here's the table of typical supported options for different kinds of nodes:

| | BLE | Serial | TCP | HTTP |
|-|-|-|-|-|
| esp32 (e.g. Heltec v3, Lilygo T3S3, T-Deck, T-Beam) | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| nrf52 (e.g. RAK4631, Heltec T114) | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: * | :x: |
| rp2040 (e.g. Raspberry Pi Pico/Pico W) | :x: | :heavy_check_mark:** | :heavy_check_mark: ** | :x: |

(* nrf52 can do TCP only with an Ethernet module)
(** rp2040 can do TCP only on nodes supporting WiFi (e.g. the Pico W), and there is a special, advanced-user "slowclock" version that also disables USB serial)

**Comment by ckambiselis** [comment]
Hmmm, so I guess I have to test running the Android app on Bluestacks or something. Forgot to say, the node is a RAK4631 with the Ethernet PoE module.

**Reply by ckambiselis** [reply]
This seems to work, I was able to connect to it using its IP address.

**Comment by iuvi7** [comment]
<img width="639" height="599" alt="изображение" src="https://github.com/user-attachments/assets/0be9e49f-1e1e-4e1d-a6ff-b92dafaa4709" />

@ckambiselis @ianmcorvidae 

Hi! Am i get it right, with such combination i can only power NRF device, but i can not send\receive messages and control node from my notebook\router (plugging RJ45 straight without white power block) ? There is no GUI over CLI tools available at the moment? 

Sorry for disturb , hope for answer

Thanks

**Reply by ckambiselis** [reply]
The black adapter you have just takes the power from the PoE feed on the ethernet cable and outputs a USB connector for power only, and the data in on the ethernet connector. My setup consists of a RAK19007 base with the RAK13800 ethernet and the RAK19018 PoE add-on for it, this way the ethernet cable is directly connected to the RAK and provides power and data connection (no need for the black adapter). 

The NRF cannot provide a GUI directly, but with my setup, if your phone is connected on the same network, you can connect to it through the RAKs IP address, you can also connect through the https://client.meshtastic.org/ website on a computer, but I find it really buggy, installing Bluestacks (Android emulator) and installing the Meshtastic app there, will give you a great GUI that works really well.
